portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,99 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import java.util.ArrayList;
+import java.util.List;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private class Comment{
+    String comment;
+    String name;
+
+    public Comment(String c, String n){
+        comment = c;
+        name = n;
+    }
+  }
+
+  private List<Comment> comments;
+
+  @Override
+  public void init() {
+        
+    comments = new ArrayList<>();
+    //comments.add(new Comment("HEY","JOSE"));
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    Query query = new Query("Comment");
+    //addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String comment = (String) entity.getProperty("comment");
+      String name  = (String) entity.getProperty("name");
+
+
+      Comment task = new Comment(comment,name);
+      comments.add(task);
+    }
+
+    String json = convertToJsonUsingGson(comments);
+
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  
+
+  private String convertToJsonUsingGson(List<Comment> serverStats) {
+    Gson gson = new Gson();
+    String json = gson.toJson(serverStats);
+    return json;
+}
+
+
+@Override
+public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    String comment = request.getParameter("comment-input");
+    String name = request.getParameter("name-input");
+
+    Comment newComment = new Comment(name,comment);
+   // String json = convertToJsonUsingGson(newComment);
+   // comments.add(newComment);
+
     response.setContentType("text/html;");
```

Comment:
This isn't needed when sending a redirect, only when writing to the output stream directly.